pkg/endpoint/policy.go

```diff
@@ -323,7 +319,7 @@ func (e *Endpoint) computeDesiredL3PolicyMapEntries(owner Owner, identityCache *
 
 	// Only L3 (label-based) policy apply.
 	// Complexity increases linearly by the number of identities in the map.
-	for identity, labels := range *identityCache {
+	for identity, labels := range *e.prevIdentityCache {
```

Comment:
I was a bit puzzled here at first. It's not obvious why prevIdentityCache is pointing to the right cache at first. Reading the rest of the code, it is but this may break as it's not obvious that the `e.prevIdentitycache` assignment cannot be moved around. I recommend to add a comment to the assignment to make sure we don't accidentally break this.